tests/test_batch.py

```diff
@@ -41,10 +41,14 @@ def test_create_batch(self, emulator):
         assert batch.jobs[self.job_id].batch_id == batch.id
         assert batch.jobs[self.job_id].runs == self.n_job_runs
 
-    def test_create_batch_and_wait(self, request_mock):
+    @pytest.mark.parametrize("wait,fetch_results", [(True, False), (False, True)])
+    def test_create_batch_and_wait(self, request_mock, wait, fetch_results):
         job = {"runs": self.n_job_runs, "variables": self.job_variables}
         batch = self.sdk.create_batch(
-            serialized_sequence=self.pulser_sequence, jobs=[job], wait=True
+            serialized_sequence=self.pulser_sequence,
+            jobs=[job],
+            wait=wait,
+            fetch_results=fetch_results,
```

Comment:
Ah sorry I commented too quickly, this covers the `fetch_result` case. So I guess we can ignore my comment about testing.